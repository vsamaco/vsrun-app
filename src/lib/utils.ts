import { type ClassValue, clsx } from "clsx";
import { type NextRequest } from "next/server";
import { twMerge } from "tailwind-merge";

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}

export const parse = (req: NextRequest) => {
  let domain = req.headers.get("host") as string;
  domain = domain.replace("www.", ""); // remove www. from domain
  if (domain === "dub.localhost:8888" || domain.endsWith(".vercel.app")) {
    // for local development and preview URLs
    domain = "dub.sh";
  }

  // path is the path of the URL (e.g. dub.co/stats/github -> /stats/github)
  const path = req.nextUrl.pathname;

  // fullPath is the full URL path (along with search params)
  const searchParams = req.nextUrl.searchParams.toString();
  const fullPath = `${path}${
    searchParams.length > 0 ? `?${searchParams}` : ""
  }`;

  // Here, we are using decodeURIComponent to handle foreign languages like Hebrew
  const key = decodeURIComponent(path.split("/")[1]); // key is the first part of the path (e.g. dub.co/stats/github -> stats)
  const fullKey = decodeURIComponent(path.slice(1)); // fullKey is the full path without the first slash (to account for multi-level subpaths, e.g. dub.sh/github/repo -> github/repo)

  return { domain, path, fullPath, key, fullKey };
};
