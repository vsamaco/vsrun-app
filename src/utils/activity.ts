import dayjs from "dayjs";
import duration from "dayjs/plugin/duration";
import relativeTime from "dayjs/plugin/relativeTime";

dayjs.extend(duration);
dayjs.extend(relativeTime);

export const formatSeconds = (timeSeconds: number) => {
  return dayjs.duration(timeSeconds, "seconds").format("H:mm");
};

export const formatDate = (isoString: string) => {
  return dayjs(isoString).format("MMM D, YYYY");
};

export const formatTime = (isoString: string) => {
  return dayjs(isoString).format("h:mm a");
};

export const formatHumanizeSeconds = (seconds: number) => {
  return dayjs.duration(seconds, "seconds").format("H[h] m[m]");
};

export const metersToMiles = (i: number) => {
  return Math.round(i * 0.000621371192 * 100) / 100;
};

export const metersToFeet = (i: number) => {
  return i * 3.2808;
};
