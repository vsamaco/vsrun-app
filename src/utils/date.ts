export function formatDate(date: Date, options?: Intl.DateTimeFormatOptions) {
  // undefined - default locale
  return new Intl.DateTimeFormat(undefined, options).format(date);
}
