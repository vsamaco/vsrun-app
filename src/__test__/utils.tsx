import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render, type RenderOptions } from "@testing-library/react";
import { createTRPCReact, httpLink } from "@trpc/react-query";
import { createTRPCMsw } from "msw-trpc";
import { type ReactElement } from "react";
import superjson from "superjson";
import fetch from "cross-fetch";
import { type AppRouter } from "~/server/api/root";
import { setupServer } from "msw/lib/node";

const mockedTRPC = createTRPCReact<AppRouter>({
  unstable_overrides: {
    useMutation: {
      async onSuccess(opts) {
        await opts.originalFn();
        await opts.queryClient.invalidateQueries();
      },
    },
  },
});

const mockedTRPCClient = mockedTRPC.createClient({
  transformer: superjson,
  links: [httpLink({ url: "http://localhost:3000/api/trpc", fetch })],
});

const mockedQueryClient = new QueryClient();

export const MockedTRPCProvider = (props: { children: React.ReactNode }) => {
  return (
    <mockedTRPC.Provider
      client={mockedTRPCClient}
      queryClient={mockedQueryClient}
    >
      <QueryClientProvider client={mockedQueryClient}>
        {props.children}
      </QueryClientProvider>
    </mockedTRPC.Provider>
  );
};

export const renderWithProviders = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => {
  return render(ui, {
    wrapper: (props) => <MockedTRPCProvider {...props} />,
    ...options,
  });
};

export const trpcMsw = createTRPCMsw<AppRouter>({
  transformer: { input: superjson, output: superjson },
});
