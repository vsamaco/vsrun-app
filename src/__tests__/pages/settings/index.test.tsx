import { cleanup, screen, waitFor, within } from "@testing-library/react";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
import { server } from "~/__tests__/server";
import { renderWithProviders, trpcMsw } from "~/__tests__/utils";
import GeneralSettingsPage from "~/pages/settings";

describe("settings", () => {
  beforeEach(() => {
    vi.mock("next-auth/react", () => {
      const mockSession = {
        expires: new Date(Date.now() + 2 * 86400).toISOString(),
        user: { username: "admin" },
      };
      return {
        __esModule: true,
        useSession: vi.fn(() => {
          return { data: mockSession, status: "authenticated" };
        }),
      };
    });
  });

  afterEach(() => {
    vi.restoreAllMocks();
    cleanup();
  });

  describe("with runprofile", () => {
    beforeEach(() => {
      server.use(
        trpcMsw.runProfile.getUserProfile.query((_input) => {
          return {
            id: "123",
            userId: "456",
            name: "Foo Bar",
            slug: "foobar",
            shoeRotations: [
              {
                id: "shoe-rotation-id",
                slug: "rotation-slug",
                name: "Foobar Rotation",
                description: "",
                startDate: new Date("2024-05-01T12:00:00Z"),
                shoes: [],
                shoeList: [],
                metadata: null,
                updatedAt: new Date("2024-05-01T12:00:00Z"),
                createdAt: new Date("2024-05-01T12:00:00Z"),
                profileId: "1",
              },
            ],
            shoes2: [
              {
                id: "shoe-id",
                slug: "foobar-shoe",
                brand_name: "brand",
                model_name: "model",
                description: "",
                distance: 1600,
                start_date: new Date("2024-05-01T12:00:00Z"),
                categories: ["daily-trainer"],
                metadata: null,
                runProfileId: "1",
                updatedAt: new Date("2024-05-01T12:00:00Z"),
                createdAt: new Date("2024-05-01T12:00:00Z"),
              },
            ],
            user: {
              accounts: [{ provider: "strava", providerAccountId: "123" }],
            },
            races: [
              {
                id: "123",
                name: "foobar race",
                slug: "foobar-race",
                description: "",
                start_date: new Date("2024-05-01T12:00:00Z"),
                start_latlng: [1, 1],
                end_latlng: null,
                workout_type: null,
                distance: 1600,
                moving_time: 3600,
                total_elevation_gain: 0,
                laps: [],
                summary_polyline: "",
                shoeId: null,
                metadata: null,
                raceProfileId: null,
                highlightRunProfileId: "1",
                createdAt: new Date("2024-05-01T12:00:00Z"),
                updatedAt: new Date("2024-05-01T12:00:00Z"),
              },
            ],
            highlight_run: {
              id: "123",
              name: "foobar run",
              slug: "foobar-run",
              description: "",
              start_date: new Date("2024-05-01T12:00:00Z"),
              start_latlng: [1, 1],
              end_latlng: null,
              workout_type: null,
              distance: 1600,
              moving_time: 3600,
              total_elevation_gain: 0,
              laps: [],
              summary_polyline: "",
              shoeId: null,
              metadata: null,
              raceProfileId: null,
              highlightRunProfileId: "1",
              createdAt: new Date("2024-05-01T12:00:00Z"),
              updatedAt: new Date("2024-05-01T12:00:00Z"),
            },
            weekStats: {
              start_date: new Date("2024-05-01T12:00:00Z"),
              end_date: new Date("2024-05-01T12:00:00Z"),
              total_runs: 1,
              total_duration: 3600,
              total_distance: 1600,
              total_elevation: 0,
              activities: [],
              metadata: null,
            },
            events: [],
            highlightRun: {},
            shoes: [],
            createdAt: new Date("2024-05-01T12:00:00Z"),
            updatedAt: new Date("2024-05-01T12:00:00Z"),
          };
        })
      );
    });

    it("renders page with profile", async () => {
      const container = renderWithProviders(<GeneralSettingsPage />);
      await waitFor(() => {
        expect(screen.getByRole("heading", { name: /foo bar/i })).toBeDefined();

        const highlightRunCard = screen.getByTestId(
          "settings-highlight-run-card"
        );
        expect(
          within(highlightRunCard).getByRole("heading", { name: /run/i })
        ).toBeDefined();
        expect(within(highlightRunCard).getByText(/foobar run/i)).toBeDefined();

        const weekStatsCard = screen.getByTestId("settings-weekstats-card");
        expect(
          within(weekStatsCard).getByRole("heading", { name: /week/i })
        ).toBeDefined();
        expect(within(weekStatsCard).getByText(/1h 0m/)).toBeDefined();

        const shoesCard = screen.getByTestId("settings-shoes-card");
        expect(
          within(shoesCard).getByRole("heading", { name: /shoes/i })
        ).toBeDefined();
        expect(within(shoesCard).getByText(/brand model/i)).toBeDefined();

        const shoeRotationsCard = screen.getByTestId(
          "settings-shoe-rotations-card"
        );
        expect(
          within(shoeRotationsCard).getByRole("heading", {
            name: /shoe rotations/i,
          })
        ).toBeDefined();
        expect(
          within(shoeRotationsCard).getByText(/foobar rotation/i)
        ).toBeDefined();

        const racesCard = screen.getByTestId("settings-races-card");
        expect(
          within(racesCard).getByRole("heading", { name: /races/i })
        ).toBeDefined();
        expect(within(racesCard).getByText(/foobar race/)).toBeDefined();
      });
      expect(container).toMatchSnapshot();
    });
  });

  describe("without runprofile", () => {
    beforeEach(() => {
      server.use(
        trpcMsw.runProfile.getUserProfile.query((_input) => {
          return null;
        })
      );
    });

    it("renders placeholder create profile", async () => {
      const container = renderWithProviders(<GeneralSettingsPage />);
      await waitFor(() => {
        expect(
          container.getByText(/You don't have a profile yet/)
        ).toBeDefined();
        expect(
          container.getByRole("button", { name: "Create Profile" })
        ).toBeDefined();
      });
    });
  });
});
