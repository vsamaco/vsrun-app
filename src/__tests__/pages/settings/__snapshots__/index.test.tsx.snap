// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`settings > with runprofile > renders page with profile 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="space-y-6"
      >
        <div
          class="mb-10 flex w-full flex-col items-center justify-between space-y-2 md:flex-row"
        >
          <div
            class="flex items-center space-x-4"
          >
            <span
              class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
            >
              <span
                class="flex h-full w-full items-center justify-center rounded-full bg-muted"
              >
                F
              </span>
            </span>
            <h3
              class=" text-lg"
            >
              Foo Bar
            </h3>
          </div>
          <div
            class="space-x-2"
          >
            <button
              aria-controls="radix-:r0:"
              aria-expanded="false"
              aria-haspopup="dialog"
              class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
              data-state="closed"
              type="button"
            >
              Edit Profile
            </button>
            <a
              href="/p/foobar"
            >
              <button
                class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
              >
                View Profile
              </button>
            </a>
          </div>
        </div>
        <div
          class="shrink-0 bg-border h-[1px] w-full"
          data-orientation="horizontal"
          role="none"
        />
        <div
          class="grid items-start justify-center gap-6 rounded-lg lg:grid-cols-2"
        >
          <div
            class="col-span-2 grid items-start gap-6 lg:col-span-1"
          >
            <div
              class="flex items-center justify-center [&>div]:w-full"
            >
              <div
                class="rounded-lg border bg-card text-card-foreground shadow-sm"
                data-testid="settings-highlight-run-card"
              >
                <div
                  class="flex flex-col space-y-1.5 p-6"
                >
                  <h3
                    class="text-2xl font-semibold leading-none tracking-tight"
                  >
                    Run
                  </h3>
                  <p
                    class="text-sm text-muted-foreground"
                  >
                    Highlight your recent activity
                  </p>
                </div>
                <div
                  class="p-6 pt-0 grid gap-4"
                >
                  <div
                    class="grid-cols mb-4 grid items-start pb-4 last:mb-0 last:pb-0"
                  >
                    <div
                      class="flex items-center justify-between space-x-4"
                    >
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="text-sm font-medium leading-none"
                        >
                          foobar run
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          May 1, 2024
                        </p>
                      </div>
                      <p
                        class="text-sm text-muted-foreground"
                      >
                        0.99
                         mi
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="flex items-center p-6 pt-0"
                >
                  <a
                    class="w-full inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                    href="/settings/run"
                  >
                    Edit Run
                  </a>
                </div>
              </div>
            </div>
            <div
              class="flex items-center justify-center [&>div]:w-full"
            >
              <div
                class="rounded-lg border bg-card text-card-foreground shadow-sm"
                data-testid="settings-weekstats-card"
              >
                <div
                  class="flex flex-col space-y-1.5 p-6"
                >
                  <h3
                    class="text-2xl font-semibold leading-none tracking-tight"
                  >
                    Week
                  </h3>
                  <p
                    class="text-sm text-muted-foreground"
                  >
                    Highlight your weekly stats
                  </p>
                </div>
                <div
                  class="p-6 pt-0 grid gap-1"
                >
                  <div
                    class="-mx-2 flex items-start space-x-4 rounded-md p-2 transition-all hover:bg-accent hover:text-accent-foreground"
                  >
                    <div
                      class="space-y-1"
                    >
                      <p
                        class="text-sm font-medium leading-none"
                      >
                        Date
                      </p>
                      <p
                        class="text-sm text-muted-foreground"
                      >
                        05/01
                         -
                         
                        05/01
                      </p>
                    </div>
                  </div>
                  <div
                    class="-mx-2 flex items-start space-x-4 rounded-md p-2 transition-all hover:bg-accent hover:text-accent-foreground"
                  >
                    <div
                      class="space-y-1"
                    >
                      <p
                        class="text-sm font-medium leading-none"
                      >
                        Distance
                      </p>
                      <p
                        class="text-sm text-muted-foreground"
                      >
                        0.99
                         mi
                      </p>
                    </div>
                  </div>
                  <div
                    class="-mx-2 flex items-start space-x-4 rounded-md p-2 transition-all hover:bg-accent hover:text-accent-foreground"
                  >
                    <div
                      class="space-y-1"
                    >
                      <p
                        class="text-sm font-medium leading-none"
                      >
                        Time
                      </p>
                      <p
                        class="text-sm text-muted-foreground"
                      >
                        1h 0m
                      </p>
                    </div>
                  </div>
                  <div
                    class="-mx-2 flex items-start space-x-4 rounded-md p-2 transition-all hover:bg-accent hover:text-accent-foreground"
                  >
                    <div
                      class="space-y-1"
                    >
                      <p
                        class="text-sm font-medium leading-none"
                      >
                        Elevation
                      </p>
                      <p
                        class="text-sm text-muted-foreground"
                      >
                        0
                         ft
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="flex items-center p-6 pt-0"
                >
                  <a
                    class="w-full inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                    href="/settings/week"
                  >
                    Edit
                     Week
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-span-2 grid items-start gap-6 lg:col-span-1"
          >
            <div
              class="flex items-center justify-center [&>div]:w-full"
            >
              <div
                class="rounded-lg border bg-card text-card-foreground shadow-sm"
                data-testid="settings-shoes-card"
              >
                <div
                  class="flex flex-col space-y-1.5 p-6"
                >
                  <h3
                    class="text-2xl font-semibold leading-none tracking-tight"
                  >
                    Shoes
                  </h3>
                  <p
                    class="text-sm text-muted-foreground"
                  >
                    Highlight your shoes
                  </p>
                </div>
                <div
                  class="p-6 pt-0 grid gap-4"
                >
                  <div
                    class="flex items-center justify-between space-x-4"
                  >
                    <div
                      class="flex items-center space-x-4"
                    >
                      <p
                        class="w-[160px] truncate text-sm font-medium leading-none"
                      >
                        brand
                         
                        model
                      </p>
                    </div>
                    <div
                      class="flex items-center space-x-4"
                    >
                      <p
                        class="text-sm text-muted-foreground"
                      >
                        1
                         mi
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="flex items-center p-6 pt-0"
                >
                  <a
                    class="w-full inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                    href="/settings/shoes"
                  >
                    Edit Shoes
                  </a>
                </div>
              </div>
            </div>
            <div
              class="flex items-center justify-center [&>div]:w-full"
            >
              <div
                class="rounded-lg border bg-card text-card-foreground shadow-sm"
                data-testid="settings-shoe-rotations-card"
              >
                <div
                  class="flex flex-col space-y-1.5 p-6"
                >
                  <h3
                    class="text-2xl font-semibold leading-none tracking-tight"
                  >
                    Shoe Rotations
                  </h3>
                  <p
                    class="text-sm text-muted-foreground"
                  >
                    Highlight your shoe rotation
                  </p>
                </div>
                <div
                  class="p-6 pt-0 grid gap-4"
                >
                  <div
                    class="flex items-center justify-between space-x-4"
                  >
                    <div
                      class="flex items-center space-x-4"
                    >
                      <p
                        class="w-[160px] truncate text-sm font-medium leading-none"
                      >
                        Foobar Rotation
                      </p>
                    </div>
                    <div
                      class="flex items-center space-x-4"
                    >
                      <p
                        class="text-sm text-muted-foreground"
                      >
                        0
                         shoes
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="flex items-center p-6 pt-0"
                >
                  <a
                    class="w-full inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                    href="/settings/shoe_rotations"
                  >
                    Edit Rotations
                  </a>
                </div>
              </div>
            </div>
            <div
              class="flex items-center justify-center [&>div]:w-full"
            >
              <div
                class="rounded-lg border bg-card text-card-foreground shadow-sm"
                data-testid="settings-races-card"
              >
                <div
                  class="flex flex-col space-y-1.5 p-6"
                >
                  <h3
                    class="text-2xl font-semibold leading-none tracking-tight"
                  >
                    Races
                  </h3>
                  <p
                    class="text-sm text-muted-foreground"
                  >
                    Highlight your races
                  </p>
                </div>
                <div
                  class="p-6 pt-0 grid gap-4"
                >
                  <div
                    class="flex items-center justify-between space-x-4"
                  >
                    <div
                      class="flex items-center space-x-4"
                    >
                      <p
                        class="w-[160px] truncate text-sm font-medium leading-none"
                      >
                        foobar race
                      </p>
                    </div>
                    <div
                      class="flex items-center space-x-4"
                    >
                      <p
                        class="text-sm text-muted-foreground"
                      >
                        0.99
                         mi
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="flex items-center p-6 pt-0"
                >
                  <a
                    class="w-full inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                    href="/settings/races"
                  >
                    Edit Races
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="space-y-6"
    >
      <div
        class="mb-10 flex w-full flex-col items-center justify-between space-y-2 md:flex-row"
      >
        <div
          class="flex items-center space-x-4"
        >
          <span
            class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
          >
            <span
              class="flex h-full w-full items-center justify-center rounded-full bg-muted"
            >
              F
            </span>
          </span>
          <h3
            class=" text-lg"
          >
            Foo Bar
          </h3>
        </div>
        <div
          class="space-x-2"
        >
          <button
            aria-controls="radix-:r0:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
            data-state="closed"
            type="button"
          >
            Edit Profile
          </button>
          <a
            href="/p/foobar"
          >
            <button
              class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
            >
              View Profile
            </button>
          </a>
        </div>
      </div>
      <div
        class="shrink-0 bg-border h-[1px] w-full"
        data-orientation="horizontal"
        role="none"
      />
      <div
        class="grid items-start justify-center gap-6 rounded-lg lg:grid-cols-2"
      >
        <div
          class="col-span-2 grid items-start gap-6 lg:col-span-1"
        >
          <div
            class="flex items-center justify-center [&>div]:w-full"
          >
            <div
              class="rounded-lg border bg-card text-card-foreground shadow-sm"
              data-testid="settings-highlight-run-card"
            >
              <div
                class="flex flex-col space-y-1.5 p-6"
              >
                <h3
                  class="text-2xl font-semibold leading-none tracking-tight"
                >
                  Run
                </h3>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Highlight your recent activity
                </p>
              </div>
              <div
                class="p-6 pt-0 grid gap-4"
              >
                <div
                  class="grid-cols mb-4 grid items-start pb-4 last:mb-0 last:pb-0"
                >
                  <div
                    class="flex items-center justify-between space-x-4"
                  >
                    <div
                      class="space-y-1"
                    >
                      <p
                        class="text-sm font-medium leading-none"
                      >
                        foobar run
                      </p>
                      <p
                        class="text-sm text-muted-foreground"
                      >
                        May 1, 2024
                      </p>
                    </div>
                    <p
                      class="text-sm text-muted-foreground"
                    >
                      0.99
                       mi
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="flex items-center p-6 pt-0"
              >
                <a
                  class="w-full inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                  href="/settings/run"
                >
                  Edit Run
                </a>
              </div>
            </div>
          </div>
          <div
            class="flex items-center justify-center [&>div]:w-full"
          >
            <div
              class="rounded-lg border bg-card text-card-foreground shadow-sm"
              data-testid="settings-weekstats-card"
            >
              <div
                class="flex flex-col space-y-1.5 p-6"
              >
                <h3
                  class="text-2xl font-semibold leading-none tracking-tight"
                >
                  Week
                </h3>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Highlight your weekly stats
                </p>
              </div>
              <div
                class="p-6 pt-0 grid gap-1"
              >
                <div
                  class="-mx-2 flex items-start space-x-4 rounded-md p-2 transition-all hover:bg-accent hover:text-accent-foreground"
                >
                  <div
                    class="space-y-1"
                  >
                    <p
                      class="text-sm font-medium leading-none"
                    >
                      Date
                    </p>
                    <p
                      class="text-sm text-muted-foreground"
                    >
                      05/01
                       -
                       
                      05/01
                    </p>
                  </div>
                </div>
                <div
                  class="-mx-2 flex items-start space-x-4 rounded-md p-2 transition-all hover:bg-accent hover:text-accent-foreground"
                >
                  <div
                    class="space-y-1"
                  >
                    <p
                      class="text-sm font-medium leading-none"
                    >
                      Distance
                    </p>
                    <p
                      class="text-sm text-muted-foreground"
                    >
                      0.99
                       mi
                    </p>
                  </div>
                </div>
                <div
                  class="-mx-2 flex items-start space-x-4 rounded-md p-2 transition-all hover:bg-accent hover:text-accent-foreground"
                >
                  <div
                    class="space-y-1"
                  >
                    <p
                      class="text-sm font-medium leading-none"
                    >
                      Time
                    </p>
                    <p
                      class="text-sm text-muted-foreground"
                    >
                      1h 0m
                    </p>
                  </div>
                </div>
                <div
                  class="-mx-2 flex items-start space-x-4 rounded-md p-2 transition-all hover:bg-accent hover:text-accent-foreground"
                >
                  <div
                    class="space-y-1"
                  >
                    <p
                      class="text-sm font-medium leading-none"
                    >
                      Elevation
                    </p>
                    <p
                      class="text-sm text-muted-foreground"
                    >
                      0
                       ft
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="flex items-center p-6 pt-0"
              >
                <a
                  class="w-full inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                  href="/settings/week"
                >
                  Edit
                   Week
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-span-2 grid items-start gap-6 lg:col-span-1"
        >
          <div
            class="flex items-center justify-center [&>div]:w-full"
          >
            <div
              class="rounded-lg border bg-card text-card-foreground shadow-sm"
              data-testid="settings-shoes-card"
            >
              <div
                class="flex flex-col space-y-1.5 p-6"
              >
                <h3
                  class="text-2xl font-semibold leading-none tracking-tight"
                >
                  Shoes
                </h3>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Highlight your shoes
                </p>
              </div>
              <div
                class="p-6 pt-0 grid gap-4"
              >
                <div
                  class="flex items-center justify-between space-x-4"
                >
                  <div
                    class="flex items-center space-x-4"
                  >
                    <p
                      class="w-[160px] truncate text-sm font-medium leading-none"
                    >
                      brand
                       
                      model
                    </p>
                  </div>
                  <div
                    class="flex items-center space-x-4"
                  >
                    <p
                      class="text-sm text-muted-foreground"
                    >
                      1
                       mi
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="flex items-center p-6 pt-0"
              >
                <a
                  class="w-full inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                  href="/settings/shoes"
                >
                  Edit Shoes
                </a>
              </div>
            </div>
          </div>
          <div
            class="flex items-center justify-center [&>div]:w-full"
          >
            <div
              class="rounded-lg border bg-card text-card-foreground shadow-sm"
              data-testid="settings-shoe-rotations-card"
            >
              <div
                class="flex flex-col space-y-1.5 p-6"
              >
                <h3
                  class="text-2xl font-semibold leading-none tracking-tight"
                >
                  Shoe Rotations
                </h3>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Highlight your shoe rotation
                </p>
              </div>
              <div
                class="p-6 pt-0 grid gap-4"
              >
                <div
                  class="flex items-center justify-between space-x-4"
                >
                  <div
                    class="flex items-center space-x-4"
                  >
                    <p
                      class="w-[160px] truncate text-sm font-medium leading-none"
                    >
                      Foobar Rotation
                    </p>
                  </div>
                  <div
                    class="flex items-center space-x-4"
                  >
                    <p
                      class="text-sm text-muted-foreground"
                    >
                      0
                       shoes
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="flex items-center p-6 pt-0"
              >
                <a
                  class="w-full inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                  href="/settings/shoe_rotations"
                >
                  Edit Rotations
                </a>
              </div>
            </div>
          </div>
          <div
            class="flex items-center justify-center [&>div]:w-full"
          >
            <div
              class="rounded-lg border bg-card text-card-foreground shadow-sm"
              data-testid="settings-races-card"
            >
              <div
                class="flex flex-col space-y-1.5 p-6"
              >
                <h3
                  class="text-2xl font-semibold leading-none tracking-tight"
                >
                  Races
                </h3>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Highlight your races
                </p>
              </div>
              <div
                class="p-6 pt-0 grid gap-4"
              >
                <div
                  class="flex items-center justify-between space-x-4"
                >
                  <div
                    class="flex items-center space-x-4"
                  >
                    <p
                      class="w-[160px] truncate text-sm font-medium leading-none"
                    >
                      foobar race
                    </p>
                  </div>
                  <div
                    class="flex items-center space-x-4"
                  >
                    <p
                      class="text-sm text-muted-foreground"
                    >
                      0.99
                       mi
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="flex items-center p-6 pt-0"
              >
                <a
                  class="w-full inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                  href="/settings/races"
                >
                  Edit Races
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
